velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -198,5 +202,28 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, dateSub) {
+  const auto dateSub = [&](std::optional<int32_t> date,
+                           std::optional<int32_t> value) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_sub(c0, c1)", {date, value}, {DATE(), INTEGER()});
+  };
+
+  // Check simple tests.
+  int32_t inputDate = parseDate("2019-03-01");
+  int32_t expectedDate = parseDate("2019-02-28");
+  EXPECT_EQ(expectedDate, dateSub(inputDate, 1));
+
+  // Account for the last day of a year-month.
+  inputDate = parseDate("2020-02-29");
+  expectedDate = parseDate("2019-01-30");
+  EXPECT_EQ(expectedDate, dateSub(inputDate, 395));
+
+  // Check for negative intervals.
+  inputDate = parseDate("2019-02-28");
+  expectedDate = parseDate("2020-02-29");
+  EXPECT_EQ(expectedDate, dateSub(inputDate, -366));
```

Comment:
What happens if you pass very large positive/negative number as input?

Reply:
Added integer overflow check of int32_t value to DateSubFunction in DateTimeFunctions.h. Kindly help to review again. Thanks.